smk/include/config_parser.smk

```diff
@@ -8,12 +8,246 @@ Authors: Carmen Saenz, Mani Arumugam
+    'MERGE_memory' : int,
+    'MERGE_threads' : int,
+    'METADATA' : 'file',
+    'METAFLYE_presets' : dict,
+    'metaphlan_version' : str,
+    'METASPADES_custom_build' : 'file',
+    'METASPADES_hybrid_max_k' : int,
+    'METASPADES_illumina_max_k' : int,
+    'METASPADES_memory' : int,
+    'METASPADES_qoffset' : str,
+    'METASPADES_threads' : int,
+    'MIN_FASTA_LENGTH' : int,
+    'MIN_MAG_LENGTH' : int,
+    'MIN_mapped_reads' : int,
+    'minto_dir' : 'directory',
+    'MINTO_MODE' : str,
+    'motus_version' : str,
+    'msamtools_filter_length' : int,
+    'MULTIPLEX_TECH' : str,
+    'NAME_host_genome' : str,
+    'NAME_reference' : str,
+    'NANOPORE' : list,
+    'omics' : str,
+    'PATH_host_genome' : 'directory',
+    'PATH_reference' : 'directory',
+    'perc_remaining_reads' : int,
+    'PHYLOPHLAN_TAXONOMY_VERSION' : str,
+    'PLOT_factor2' : str,
+    'PLOT_time' : str,
+    'PROJECT' : str,
+    'raw_reads_dir' : 'directory',
+    'READ_minlen' : int,
+    'rRNA_index_memory' : int,
+    'rRNA_index_threads' : int,
+    'RUN_TAXONOMY' : bool,
+    'SAMTOOLS_sort_perthread_memgb' : int,
+    'SAMTOOLS_sort_threads' : int,
+    'SCORE_METHOD' : str,
+    'sortmeRNA_memory' : int,
+    'sortmeRNA_threads' : int,
+    'sortmeRNA_db' : 'directory',
+    'sortmeRNA_db_idx' : 'directory',
+    'SOURMASH_cutoff' : float,
+    'SOURMASH_max_abund' : int,
+    'SOURMASH_min_abund' : int,
+    'SPADES_CONTIGS_OR_SCAFFOLDS' : str,
+    'TAXA_memory' : int,
+    'TAXA_profiler' : str,
+    'TAXA_threads' : int,
+    'TAXONOMY_CPUS' : int,
+    'TAXONOMY_memory' : int,
+    'TAXONOMY_NAME' : str,
+    'TRIMMOMATIC_adaptors' : str,
+    'TRIMMOMATIC_index_barcodes' : str,
+    'TRIMMOMATIC_memory' : int,
+    'TRIMMOMATIC_palindrome' : 'file',
+    'TRIMMOMATIC_simple_clip_threshold' : int,
+    'TRIMMOMATIC_threads' : int,
+    'VAMB_GPU' : bool,
+    'VAMB_memory' : int,
+    'VAMB_THREADS' : int,
+    'working_dir' : 'directory'
+}
+
+# Aliases for keys.
+# E.g., This module will always refer to minimum read length with 'READ_minlen'.
+#       But if it cannot find 'READ_minlen', it will look for 'TRIMMOMATIC_minlen'.
+ALIASES = {
+    'READ_minlen' : ['TRIMMOMATIC_minlen'],
+    'FASTP_adapters' : ['FASTP_adaptors'],
+    'TRIMMOMATIC_adaptors' : ['TRIMMOMATIC_adapters']
+}
+
+##############################################
+# Functions to parse the keys
+##############################################
+
+def validate_keytype(config, type_dict, key):
+    # Get all possible aliases
+    pos_keys = [key]
+    if key in ALIASES:
+        pos_keys += ALIASES[key]
+
+    # Check for key in global type registry
+    if key not in type_dict:
+        raise KeyError(f"Configuration key not found in global registry: {key}")
+
+    # Check for all aliases
+    # Get value
+    value = None
+    for x in pos_keys:
+        if x in config:
+            value = config[x]
+            break
+    if value == None:
+        return None
+
```

Comment:
I would prefer to have two different errors for `key` not in `config` and `config[key] == None`

Reply:
Good point. Will fix.